### ISSUE TITLE
On Android, <Enter> should behave same way for advanced search as for regular search

### ISSUE THREAD

**Issue by alanfgh** [Tatoeba/tatoeba2#1726]
In Issue #1708, I reported that, in Chrome on Android, hitting <Enter> when the focus is in the text field of a regular search form unsets the "From" field, whereas the user probably wants to simply submit the search. @trang fixed that behavior by setting `type = "search"` on the query input. I see the old type of behavior for advanced search, which suggests that we want to set `type = "search"` on that query input as well.

**Comment by whysokey** [comment]
I would like to work on that!

**Comment by whysokey** [comment]
Hey! I wanted to see the code of the search bar using F12 and copying a part of it to look it up in Visual Studio but it seem to not find it. What would be the easiest way to get there?

**Comment by Yorwba** [comment]
You can inspect the search bar in your browser's dev tools (e.g. in Firefox, right click inside and choose "Inspect Element"). That will show you the corresponding HTML code, in this case
```html
<input id="SentenceQuery" type="search" name="query" ng-model="ctrl.searchQuery" ng-init="ctrl.searchQuery = ''" accesskey="4" dir="auto" flex="" class="ng-pristine ng-valid flex ng-empty ng-touched" aria-invalid="false" style="" lang="">
```
The part where it says `id="SentenceQuery"` is promising, since the id should ideally be unique. A full text search over all files (I use ripgrep, but Visual Studio probably has that functionality, too) shows results in 5 files:
```
src/Template/Element/short_description.ctp
102:                        'id' => 'SentenceQuery',

src/Template/Element/search_bar.ctp
75:            <label for="SentenceQuery">
78:            <input id="SentenceQuery"

webroot/css/pages/index.css
92:#new-search-bar #SentenceQuery {

webroot/css/layouts/default.css
362:.search_bar label[for="SentenceQuery"] {
371:#SentenceQuery {

webroot/js/elements/search-bar.ctrl.js
25:            vm.searchQuery = angular.element('#SentenceQuery').data('query');
51:                angular.element('#SentenceQuery').focus();
```

**Comment by whysokey** [comment]
Found it with the first result! Also found the advanced search, I assume somewhere there the type = "search" command should be introduced but in the simple search it goes together with input, however in this case there's no input. Is it then a good place to place it?
![image](https://user-images.githubusercontent.com/63640405/83162587-39d18a00-a10a-11ea-86cf-e36a83567272.png)

**Comment by AndiPersti** [comment]
Although `src/Template/Sentences/advanced_search.ctp` is the correct view template for the `advanced_search` action, that template uses a [view element](https://book.cakephp.org/3/en/views.html#elements) to output the advanced search form:
https://github.com/Tatoeba/tatoeba2/blob/7d9c6f545059bb3b0c6fa16a81046eb06e0e4272/src/Template/Sentences/advanced_search.ctp#L26-L28
The element's name is `advanced_search_form` and following CakePHP's convention for file naming, the actual implementation of that form is in `src/Template/Element/advanced_search_form.ctp`.

To get more familiar with CakePHP and its conventions I would recommend for a start to look at [CakePHP at a glance](https://book.cakephp.org/3/en/intro.html#cakephp-at-a-glance). I've also found [Intro to CakePHP for Bug Hunters](https://medium.com/tenable-techblog/intro-to-cakephp-for-bug-hunters-97b89f8876bd) quite useful when I've started contributing.

**Comment by whysokey** [comment]
Awesome! Thanks for the resources, I will definitely take a look at PHP this weekend. 

Also, I think I found the place where they type should be modified:
![image](https://user-images.githubusercontent.com/63640405/83228731-1698f000-a187-11ea-9119-7934ea16b353.png)

Is that correct?

**Comment by AndiPersti** [comment]
No, that's the `Search` button on the bottom of the advanced search form which has to be of `type="submit"`.

There's another little difficulty. We do not explicitly implement the input control using HTML tags (i.e. `<input ...>`) but instead use a CakePHP helper function: `$this->Form->input(...)`. This function takes as the first parameter the name of the control and as the second parameter an options array. The options are given as key/value pairs and the option you want to use is unsurprisingly called `type`. The result of the function is the HTML code which will be sent to the client.

Can you find now the correct place for adding the `type`?

**Comment by whysokey** [comment]
So that would be $this->Form->input('something', array(type...)? I found the advanced_search_form.ctp file but it doesn't seem to have anything similar. And this is the same file as in my previous screenshot. 

However, I also found search.ctp file but I don't see any type array there.

**Comment by AndiPersti** [comment]
Here's the part implementing the query control:
https://github.com/Tatoeba/tatoeba2/blob/7d9c6f545059bb3b0c6fa16a81046eb06e0e4272/src/Template/Element/advanced_search_form.ctp#L38-L48

**Comment by whysokey** [comment]
Ohhh, so wait, initially there was no "type" and we have to set it all, right? I thought there was already "type = something" and we just needed to change it to "type = Search"!

So I basically need to add 'type' = Search 
after 'dir' (line 44), is that right?

**Comment by AndiPersti** [comment]
> Ohhh, so wait, initially there was no "type" and we have to set it all, right? I thought there was already "type = something" and we just needed to change it to "type = Search"!

There is a `type="text"` attribute in the resulting HTML but it is automatically added by `Form->input()` based on the first parameter.

> So I basically need to add 'type' = Search
> after 'dir' (line 44), is that right?

Correct.

**Comment by whysokey** [comment]
Alright! So... 
1. I edit it on my PC
![image](https://user-images.githubusercontent.com/63640405/83349061-1c7f0480-a332-11ea-944a-d469eccff351.png)

2. I save it and I make a pull request here on GitHub, right?

**Comment by AndiPersti** [comment]
Basically yes, but you have a syntax error in your code. Try loading the page locally: http://localhost:8080/eng/sentences/advanced_search

Look at the lines above your change to see how to set a string value to the `type` array key. You can find further information about PHP arrays in the [PHP manual](https://www.php.net/manual/en/language.types.array.php).

**Comment by whysokey** [comment]
Is this what you meant?
![image](https://user-images.githubusercontent.com/63640405/83351943-6e328980-a348-11ea-92f5-44783e57f865.png)

I set it to type = Search because I thought that's literally what is missing, I didn't know about the whole structure being there :)

**Comment by whysokey** [comment]
I'm missing a ' in controller, fixed!

**Comment by AndiPersti** [comment]
Sorry if I was not clear.

You have written
````php
'type' = Search,
````
but this won't work because in PHP, you need to use `=>` to assign values to keys and you need to enclose string literals with `'` or `"`. So your line should look like:
````php
'type' => 'search',
````

**Comment by whysokey** [comment]
Done! So now I need my own branch in order to pull a request, right? However, I seem to not be able to create one

**Comment by AndiPersti** [comment]
What did you try?

Usually you should create a branch before adding changes.

Whenever I want to work on something new I do
1. `git checkout -b <branchname> dev` to create a new branch based on the `dev` branch
2. write the code and test it
3. `git add .` to stage all my changes (or `git add <filename>` if I don't want to stage all files together)
4. `git diff --staged` to display all my changes in a diff and go back to 2. if I notice something which I want to correct
5. `git commit` to commit my changes
6. `git push -u origin <branchname>` to push the commit to GitHub. This also sets up a upstream reference so that for additional commits I simply can write `git push`

**Comment by whysokey** [comment]
So at step 5 it tells me: 
![image](https://user-images.githubusercontent.com/63640405/83505402-073adf00-a4c6-11ea-9325-9634aeebc31e.png)
but when I run of the command it suggests, it doesn't work. 
Should it be like: git config --global user.email myemail@gmail.com? Or that's incorrect?

**Comment by AndiPersti** [comment]
> but when I run of the command it suggests, it doesn't work.

What is the error message?

> Should it be like: git config --global user.email [myemail@gmail.com](mailto:myemail@gmail.com)? 

Yes, it should be as that but with your e-mail and name inside quotes.
But haven't we set your identity in the last dev call we had? What is the output of `git config --get user.name`?
From where do you run the commands? Inside the VM?

**Comment by whysokey** [comment]
Ok! I managed to do this, I created a branch called "advancedsearch", can you see it?

**Comment by AndiPersti** [comment]
I can't see it on your [active branches list](https://github.com/whysokey/tatoeba2/branches/active).
Have you pushed it to your GitHub repo (step 6)?

But I've noticed that there is already a branch [`advancedsearch`](https://github.com/whysokey/tatoeba2/tree/advancedsearch) in your repo. It was copied when you've cloned the main repo. So you should probably use another name.

What is the output of `git branch`?
If you want to rename a branch you can use `git branch -m <old name> <new name>`.
